src/NuGet.Core/NuGet.Commands/PublicAPI/net472/PublicAPI.Shipped.txt

```diff
@@ -714,7 +714,6 @@ NuGet.Commands.RestoreRequest.RestoreForceEvaluate.get -> bool
 NuGet.Commands.RestoreRequest.RestoreForceEvaluate.set -> void
 NuGet.Commands.RestoreRequest.RestoreOutputPath.get -> string
 NuGet.Commands.RestoreRequest.RestoreOutputPath.set -> void
-NuGet.Commands.RestoreRequest.RestoreRequest(NuGet.ProjectModel.PackageSpec project, NuGet.Commands.RestoreCommandProviders dependencyProviders, NuGet.Protocol.Core.Types.SourceCacheContext cacheContext, NuGet.Packaging.Signing.ClientPolicyContext clientPolicyContext, NuGet.Common.ILogger log) -> void
```

Comment:
If I'm reading this right, this is the only `RestoreRequest` constructor currently shipping, and therefore if you break it, all internal partners and external customers who are using this API will be broken.

Reply:
You are correct Andy. I added a new parameter to the existing `RestoreRequest` constructor as per https://github.com/NuGet/NuGet.Client/pull/4111#discussion_r651273013 conversation. I am okay to change the code based on the feedback.

Reply:
@zivkan 

How critical do you think it is to keep the old constructor?

How likely is it that someone upgrades their NuGet packages, yet they are not compiling against NuGet.Commands.

Reply:
After some searching on github, I see that no repo in the dotnet org doesn't use `RestoreRequest` directly, and looking more generally, it appears that customers don't create instances of the class themselves, but use `DependencyGraphSpecRequestProvider` instead, which creates the instances.  So, it's probably a low chance that customers are creating instances themselves.

Plus as Nikolche is hinting at, I guess it's unlikely that anyone is creating their own helper libraries that use NuGet.Commands. If there isn't a program using two assemblies compiled against different versions of NuGet.Commands.dll, then there's no ABI concerns, and customers will get compile time failures if they try to upgrade the package without updating their code.

I don't know if we're feeling brave enough to make the constructor internal since it appears that anyone using the package doesn't need to create their own instances. We could also consider a struct for args, since any code doing `new RestoreRequestArgs() { Prop1 = value1, Prop2 = value2 }` will be API and ABI compatible, as long as the new property is allowed to be null (and if not, if our code does a null check and uses a default value, then it will still work). By using a struct, it's no allocation, we just need to be a little bit careful not to accidentally box the struct.

Although, I'm guessing that we don't want to spend time doing non-trivial API design/improvements, so we'll probably merge this change given it appears low risk of breaking people. It's a shame so much effort is needed to determine it's an API that nobody probably uses.

Reply:
I really appreciate @zivkan's analysis on this. 

I'm comfortable with either approach, even if I'm leaning toward having the new constructor only.

Reply:
This comment can now be resolved because in https://github.com/NuGet/NuGet.Client/pull/3934 the existing constructor of `RestoreRequest` has been marked `Obsolete` and a new constructor was added with new parameters. No Public API changes in this PR.